### PR TITLE
[IRN-4252] New Bottom Sheet Component

### PR DESCRIPTION
**Pull request by nprigg**
### Why has this change bean made

Confidence Message has been running for a while in Banana with a bottom sheet, other projects in Banana are now also seeking to use a bottom sheet - ratings POC and total cost info modal. We want to add this component to backpack so:

- we can install the component from backpack
- other areas of the business can use it and deliver a consistent user experience.

The design figma for the component including props is here - https://www.figma.com/file/MmHoBc2I6J1t5jairK1tmd/Bottom-Sheet-for-web?type=design&node-id=383-7956&t=BpESm3Cu4vzaAQw0-0

### What are the changes

PR introduced the bottom sheet component.

Remember to include the following changes:

- [x] `README.md` (If you have created a new component)
- [x] Component `README.md`
- [x] Tests
- [x] Storybook examples created/updated

### CODE REVIEW

File: packages/bpk-component-bottom-sheet/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const BACKGROUND_ID = 'background';
+export const X_ID = 'x';
```
Comment (olliecurtis) [inline packages/bpk-component-bottom-sheet/src/constants.ts:20]: Small comment. Could this be changed to be `CLOSE_ID`  instead just to make it a bit clearer that is for the close button than just X?

**Review by olliecurtis — CHANGES_REQUESTED** [review]

File: packages/bpk-component-bottom-sheet/src/constants.ts (hunk 1 above)
Comment (nprigg) [reply packages/bpk-component-bottom-sheet/src/constants.ts]: Done!

File: packages/bpk-component-bottom-sheet/src/BpkBottomSheet.module.scss [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * limitations under the License.
+ */
+
+@import '~bpk-mixins/index.scss';
+
+$height: $bpk-line-height-base * 0.82;
+
+.bpk-container {
+  position: absolute;
+  z-index: 1;
```
Comment (nprigg) [inline packages/bpk-component-bottom-sheet/src/BpkBottomSheet.module.scss:25]: Accessibility have no concerns over using the z-index. https://skyscanner.slack.com/archives/C01GSKAPFB2/p1686560002652159?thread_ts=1686301427.090329&cid=C01GSKAPFB2

**Comment by anambl** [comment]
Hey @nprigg 👋 what is the status of this? I'm assuming it's not actively being worked on in which case should we close it? 😄

**Comment by nprigg** [comment]
Hey @anambl, 
We found some accessibility issues with the :banana: version and we'll need to update the BpkModal header so we don't get the bug we found on iOS, where the details header isn't dimmed.
We've got an [epic](https://skyscanner.atlassian.net/browse/IRN-4385) for this work when we have capacity.

**Comment by anambl** [comment]
> 

no problem at all, should we close this PR then and can re-open it later if needed?

**Pull request closed without merging**